simpleclient/src/main/java/io/prometheus/client/exemplars/ExemplarConfig.java

```diff
@@ -0,0 +1,78 @@
+package io.prometheus.client.exemplars;
+
+/**
+ * Static configuration class for Exemplar behavior.
+ */
+public class ExemplarConfig {
+
+  private static final ExemplarSampler noopExemplarSampler = new NoopExemplarSampler();
+
+  private static volatile boolean enabled = true;
+  private static volatile HistogramExemplarSampler histogramExemplarSampler;
+  private static volatile CounterExemplarSampler counterExemplarSampler;
+
+  static {
+    ExemplarSampler defaultExemplarSampler = new Tracer().initExemplarSampler(noopExemplarSampler);
+    counterExemplarSampler = defaultExemplarSampler;
+    histogramExemplarSampler = defaultExemplarSampler;
+  }
+
+  /**
+   * @param counterExemplarSampler will be used by default when creating new {@code Counter} metrics,
+   *                               unless {@link #disableExemplarSamplers()} was called.
+   */
+  public static void setCounterExemplarSampler(CounterExemplarSampler counterExemplarSampler) {
+    if (counterExemplarSampler == null) {
+      throw new NullPointerException();
+    }
+    ExemplarConfig.counterExemplarSampler = counterExemplarSampler;
+  }
+
+  /**
+   * @param histogramExemplarSampler will be used by default when creating new {@code Histogram} metrics,
+   *                                 unless {@link #disableExemplarSamplers()} was called.
+   */
+  public static void setHistogramExemplarSampler(HistogramExemplarSampler histogramExemplarSampler) {
+    if (histogramExemplarSampler == null) {
+      throw new NullPointerException();
+    }
+    ExemplarConfig.histogramExemplarSampler = histogramExemplarSampler;
+  }
+
+  /**
+   * Disable the implicit exemplar samplers by default.
+   * Exemplars can still be enabled for individual metrics in the metric builder,
+   * or they can be provided explicitly for individual observations with the {@code ...withExemplar()} methods.
+   */
+  public static void disableExemplarSamplers() {
+    enabled = false;
```

Comment:
This is still racey, you don't know when this could be called relative to when Counters and Histograms are instantiated.

Reply:
> I added a line to the JavaDoc saying that this must be called before a Counter or Histogram is created.

That's not possible for users to do, as metrics are usually defined statically and thus users have no control over it.

> However, in order to implement this we would need to make access to the exemplar sampler thread safe, which might severely slow down multi-threaded use of the metrics.

If you only want to read the current state of the sampler, that should always be fast as you don't need mutexes. It's only when mixing reads and writes that things can get a tad slow.